File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        st=set()
        n=len(s)
        res=0
        i=0
        for j in range(n):
            while s[j] in st:
                st.remove(s[i])
                i+=1
            st.add(s[j])
            res=max(res,j-i+1)
        return res

File: Data Structures & Algorithms/sliding-window-maximum/submission-6.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int n=nums.length;
        if (n == 0 || k == 0) return new int[0];
        int res[]=new int [n-k+1];
        int count=0;
        PriorityQueue<Integer> pq=new PriorityQueue<>((a,b)->b-a);
        for(int i=0;i<k;i++){
            pq.offer(nums[i]);
        }
        res[count++]=pq.peek();
        for(int i=k;i<n;i++){
            pq.remove(nums[i-k]);
            pq.offer(nums[i]);
            res[count++]=pq.peek();
        }
        return res;
    }
}
